packages/atlas-service/src/main.ts

```diff
@@ -72,24 +72,25 @@ export async function throwIfNotOk(
     return;
   }
 
-  let serverErrorName = 'NetworkError';
-  let serverErrorMessage = `${res.status} ${res.statusText}`;
-  // We try to parse the response to see if the server returned any information
-  // we can show a user.
-  try {
-    const messageJSON = await res.json();
-    if (isServerError(messageJSON)) {
-      serverErrorName = 'ServerError';
-      serverErrorMessage = `${messageJSON.errorCode}: ${messageJSON.detail}`;
-    }
-  } catch (err) {
-    // no-op, use the default status and statusText in the message.
-  }
+  const messageJSON: { detail?: string; errorCode?: string } = await res
```

Comment:
You sholudn't need to assert here, `isServerError` is specifically there for it (and the assertion is a bit off anyway as you are catching the error)